### PR TITLE
BGO Positioning from macro file

### PR DESCRIPTION
**Pull request by DiegoCostas97**
Now you can tune the position of the BGO in the tank. AmBeGen primaries vertex position automatically adjust to keep being launched from BGO's centre.

### CODE REVIEW

File: macros/jobOptions.mac [hunk 1]
```diff
@@ -12,8 +12,8 @@
 #
 #/WCSim/physics/list QGSP_BIC_HP #preferred for energies below 5 GeV
 #/WCSim/physics/list LBE # preferred for low-background experiments.  
-/WCSim/physics/list FTFP_BERT # recommended for high energy physics and hadron production up to beam momentum of several tens of GeV/c.
-#/WCSim/physics/list FTFP_BERT_HP
-/WCSim/physics/nCapture Default
+#/WCSim/physics/list FTFP_BERT # recommended for high energy physics and hadron production up to beam momentum of several tens of GeV/c.
+/WCSim/physics/list FTFP_BERT_HP
+#/WCSim/physics/nCapture Default
```
Comment (tdealtry) [inline macros/jobOptions.mac]: Please revert this. If you'd like to modify the default physics option for WCTE, please submit a new PR that solves #422

File: src/WCSimDetectorMessenger.cc [hunk 2]
```diff
@@ -626,6 +633,9 @@ WCSimDetectorMessenger::~WCSimDetectorMessenger()
   delete TankRadiusChange;
   delete SetPMTPositionInput;
   delete SetCDSFile;
+
+  delete BGOPlacement;
+  //delete BGOPosition;
```
Comment (tdealtry) [inline src/WCSimDetectorMessenger.cc]:
```suggestion
  delete BGOPosition;
```

**Review by tdealtry — CHANGES_REQUESTED** [review]
Thanks for this @DiegoCostas97 it looks good. Just a couple of requests

File: macros/jobOptions.mac (hunk 1 above)
Comment (DiegoCostas97) [reply macros/jobOptions.mac]: 👍

File: src/WCSimDetectorMessenger.cc (hunk 2 above)
Comment (DiegoCostas97) [reply src/WCSimDetectorMessenger.cc]: 👍

File: macros/WCTE.mac [hunk 3]
```diff
@@ -75,9 +79,9 @@
 # Good for the low energy running.
 # 4. Switch off the QE, ie. set it at 100%
 
-/WCSim/PMTQEMethod     Stacking_Only
+#/WCSim/PMTQEMethod     Stacking_Only
 #/WCSim/PMTQEMethod     Stacking_And_SensitiveDetector
-#/WCSim/PMTQEMethod     SensitiveDetector_Only
+/WCSim/PMTQEMethod     SensitiveDetector_Only
```
Comment (tdealtry) [inline macros/WCTE.mac:84]: Please confirm whether this is what WCTE want as the new default

**Review by tdealtry — COMMENTED** [review]
For some reason that I don't understand, this is causing a change in the geometry of WCTE in the validation. See for example [here](https://wcsim.github.io/Validation/e9dc1e2ed22bead34ebf826f622a85357e7eed83/23/index.html)
(It seems to be detector size change -> PMT position change -> hit times/ NHits/... changes)

I presume you don't want the geometry to change with this PR, so please try to fix it

**Comment by kmtsui** [comment]
@DiegoCostas97 Please pull from the latest develop branch to pick up all the new development

File: src/WCSimDetectorMessenger.cc [hunk 4]
```diff
@@ -125,6 +125,13 @@ WCSimDetectorMessenger::WCSimDetectorMessenger(WCSimDetectorConstruction* WCSimD
   BGOPlacement->SetParameterName("BGOPlacement", false);
   BGOPlacement->AvailableForStates(G4State_PreInit, G4State_Idle);
 
+  BGOPosition = new G4UIcmdWith3VectorAndUnit("/WCSim/BGOPosition", this);
```
Comment (kmtsui) [inline src/WCSimDetectorMessenger.cc:128]: With this new command, maybe we can just use this to activate placement of BGO and completely remove `BGOPlacement` command?

Comment (kmtsui) [reply src/WCSimDetectorMessenger.cc:128]: Although it would not work in the multi-run scenario to remove the BGO in later runs

**Comment by DiegoCostas97** [comment]
Sorry for the inconvenience. I will close this PR and make one from my develop branch with the new changes implemented.

**Pull request closed without merging**